subsys/net/lib/sockets/socketpair.c

```diff
@@ -476,19 +476,11 @@ static ssize_t spair_write(void *obj, const void *buffer, size_t count)
 				goto out;
 			}
 
-			res = k_sem_take(&remote->sem, K_NO_WAIT);
+			res = k_sem_take(&remote->sem, K_FOREVER);
```

Comment:
It's Coverity in its case 210611 said that K_NO_WAIT here needs to be replaced in K_FOREVER, and called it all "dead code"? That seems want the commit message suggests, or I don't understand it.

Reply:
Yeah, I'm not convinced this is right.  Coverity analyzed a path where this would be reached in a condition where `is_nonblock` is false (because if it wasn't it'd have jumped to `out`).  In that case the "check-no-wait, if fail check other things and if not fail wait forever" reduces to "wait forever".  I don't believe that path is the only path that reaches this point, and in others `is_nonblock` would be true.

I'd try removing that by factoring out the no-wait/check/wait-forever sequence, which appears at least twice in this function, to a separate function.  And I'd do that in a separate PR because it's messy.

Reply:
It's a bit confusing. I'll have another look soon

Reply:
@pabigot, @pfalcon - I think Coverity might actually be right in this case.

So the conditions that get is until this loop are 
1. `is_nonblock == false` (i.e. we *may* block)
1. `avail == 0` (i.e. no space to write, so we *will* block, hence `will_block == true`).

So for all intents and purposes, we are legitimately allowed to 
* wait for a signal on `&remote->read_signal` (which blocks)
* attempt to pull up the file object pointed to by the (integer) remote fd (which might be NULL, if it was deleted from under our noses)
* attempt to take a semaphore (which could take forever since we *may* block and in fact we *expect* to block)

If `avail > 0`, `will_block` is false, and a write will occur immediately (regardless of the value of `is_nonblock`), then we do not go into the loop.

However, there is a ticket open for a scenario that I had not initially checked for - which would be reading to or writing from a socket in ISR context (it sounds comical, but someone _might_ actually try to do that and doing so should always fail if the socket is blocking). That is issue #25417 which I'll get to when we're done here.

Reply:
@cfriedt: My concern, as I mentioned, is to have clear enough and detailed enough commit description. That's how I review stuff - read commit description, and compare with what code changes actually do. Any discrepancies aren't helpful. (And this review may happen now or in a year, when investigating some (un)related issue.)

Reply:
@pfalcon - commit message has been updated.